src/ruby/yast/i18n.rb

```diff
@@ -53,9 +53,14 @@ def textdomain(domain) # usually without brackets like textdomain "example"
     def _(str)
       # no textdomain configured yet
       if !@my_textdomain
-        Yast.y2warning("No textdomain configured, cannot translate #{str.inspect}")
-        Yast.y2warning("Called from: #{::Kernel.caller(1).first}")
-        return str.freeze
+        begin
+          raise "No textdomain configured in #{self.class}, " \
```

Comment:
Um, I do not like this begin - raise - rescue - raise structure at all. :worried: 

To avoid repeating the message I'd prefer something like this:

```ruby
msg = "No textdomain configured in #{self.class}, " \
"cannot translate #{str.inspect}"

raise msg if ENV["Y2STRICTTEXTDOMAIN"]

Yast.y2warning(1, "%1", msg)
return str.freeze
```

Reply:
You are right, this is better, thanks!